File: EPRO.Library.v3.5/EPRO.Library.v3.5/LikeIgnoreCaseComparer.cs
using System.Collections.Generic;

namespace ELibrary.Standard.Objects
{
    public class LikeIgnoreCaseComparer : IEqualityComparer<string>
    {
        bool IEqualityComparer<string>.Equals(string x, string y)
        {
            if (x is null)
                return false;
            return x.ToLower().IndexOf(y.ToLower()) >= 0;
        }

        public bool Equals1(string x, string y) => ((IEqualityComparer<string>)this).Equals(x, y);

        int IEqualityComparer<string>.GetHashCode(string obj)
        {
            return obj.GetHashCode();
        }

        public int GetHashCode1(string obj) => ((IEqualityComparer<string>)this).GetHashCode(obj);
    }
}

File: ELibrary.Standard.VB/LikeIgnoreCaseComparer.vb
Namespace Objects

    Public Class LikeIgnoreCaseComparer
        Implements IEqualityComparer(Of String)


        Public Function Equals1(x As String, y As String) As Boolean Implements IEqualityComparer(Of String).Equals
            If x Is Nothing Then Return False
            Return x.ToLower().IndexOf(y.ToLower()) >= 0
        End Function

        Public Function GetHashCode1(obj As String) As Integer Implements IEqualityComparer(Of String).GetHashCode
            Return obj.GetHashCode()
        End Function




    End Class


End Namespace